pytest/tests/sanity/rpc_key_value_changes.py

```diff
@@ -3,98 +3,171 @@
+#                    "receipt_hash": "7ZYAy37zjc6ectrUifocCXxgZTdXYTce1gAiXLE8vv9a",
+#                    "type": "receipt_processing"
+#                },
+#                "change": {
+#                    "key_base64": "bXlrZXk="  # it is b"mykey" in base64 encoding
+#                    "value_base64": "bXl2YWx1ZQ=="  # it is b"myvalue" in base64 encoding
+#                },
+#                "type": "data_update"
 #            }
 #        ]
 #    }
 #}
 
-tx_block_hash = res['result']['transaction_outcome']['block_hash']
+tx_block_hash = function_call_2_response['result']['transaction_outcome']['block_hash']
 tx_account_id = nodes[0].signer_key.account_id
 
-changes = nodes[0].get_changes(
-    tx_block_hash,
-    {
-        "changes_type": "data_changes",
-        "account_id": tx_account_id,
-        "key_prefix": list(b"mykey")
-    }
-)
-changes_by_key = changes['result']['changes_by_key']
-assert len(changes_by_key) == 1
-changes_by_mykey = changes_by_key[0]
-assert set(changes_by_mykey) >= {'key', 'changes'}
-assert changes_by_mykey['key'] == [109, 121, 107, 101, 121], \
-    "changed key is expected to be 'mykey' as an array of bytes, but %r found" \
-    % changes_by_mykey['key']
-value_changes = changes_by_mykey['changes']
-assert len(value_changes) == 1
-assert set(value_changes[0]['cause']) == {'ReceiptProcessing'}
-assert value_changes[0]['value'] == [109, 121, 118, 97, 108, 117, 101], \
-    "changed value is expected to be 'myvalue' as an array of bytes, but %r found" % value_changes
-for node in nodes[1:]:
-    changes_from_another_node = node.get_changes(
-        tx_block_hash,
+state_changes_request = {
+    "block_id": tx_block_hash,
+    "changes_type": "data_changes",
+    "account_id": tx_account_id,
+    "key_prefix_base64": base64.b64encode(b"my_key").decode('utf-8'),
+}
+
+expected_changes_response = {
+    "block_hash": state_changes_request["block_id"],
+    "changes": [
         {
-            "changes_type": "data_changes",
-            "account_id": tx_account_id,
-            "key_prefix": list(b"mykey")
+            'cause': {
+                'type': 'receipt_processing',
+            },
+            'type': 'data_update',
+            'change': {
+                'key_base64': base64.b64encode(b"my_key").decode('utf-8'),
+                'value_base64': base64.b64encode(b"my_value_1").decode('utf-8'),
+            }
+        },
+        {
+            'cause': {
+                'type': 'receipt_processing',
+                'receipt_hash': function_call_2_response['result']['receipts_outcome'][0]['id']
+            },
+            'type': 'data_update',
+            'change': {
+                'key_base64': base64.b64encode(b"my_key").decode('utf-8'),
+                'value_base64': base64.b64encode(b"my_value_2").decode('utf-8'),
+            }
         }
-    )
-    assert not deepdiff.DeepDiff(changes_from_another_node, changes), "query same changes gives different result"
+    ]
```

Comment:
Example of changes RPC response when two transactions in a single block cause a change in the same key.